packages/home_widget/android/src/main/kotlin/es/antonborri/home_widget/HomeWidgetPlugin.kt

```
@@ -226,6 +225,38 @@ class HomeWidgetPlugin :
+   *
+   * Returns `null` after completing [result] with [errorCode] when no name was given or no such
+   * class exists.
+   */
+  private fun resolveWidgetClass(call: MethodCall, result: Result, errorCode: String): Class<*>? {
+    val qualifiedName = call.argument<String>("qualifiedAndroidName")
+    val className = call.argument<String>("android") ?: call.argument<String>("name")
+    if (qualifiedName == null && className == null) {
+      result.error(
+          errorCode,
+          "InvalidArguments ${call.method} must be called with 'qualifiedAndroidName', 'androidName' or 'name'",
+          IllegalArgumentException(),
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)